scripts/src/manage_db_data.ts

```diff
@@ -328,36 +241,63 @@ export async function initDb(scriptConfig: ScriptConfig, closeConnectionWhenDone
 				}
 			}));
 		}
-		// create offers from csv
-		const parseCsv = require("csv-parse/lib/sync");
+		// create offers from json
 		const createOfferOptions: EarnOptions = {
 			doNotUpdateExiting: scriptConfig.no_update!,
 			dryRun: scriptConfig.dry_run!,
 			confirmUpdate: scriptConfig.require_update_confirm!,
 			onlyUpdateMetaImage: scriptConfig.update_earn_thumbnails,
 			onlyUpdate: scriptConfig.only_update,
 		};
-		for (const filename of fs.readdirSync(path(offersDir))) {
-			console.log(`read ${ filename }`);
+		type OfferData = {
+			id: string;
+			name: string;
+			amount: number;
+			type: "earn" | "spend";
+			meta: string;
+		};
+		type OfferDataMeta = {
+			title: string,
+			image: string,
+			description: string,
+			order_meta: { title: "Poll" | "Quiz", description: "Completed" }
+		};
 
-			const offersCsv = fs.readFileSync(path(join(offersDir, filename)));
-			const parsed = parseCsv(offersCsv);
+		type OfferContentData = {
+			id: string;
+			offer_id: string;
+			content: string;
+			content_type: ContentType;
+		};
 
-			const title = readTitle(parsed[0][0]);
-			const contentType = parsed[0][0].split(/ +/, 2)[1].toLowerCase() as ContentType;
-			let results = [];
-			if (title === "Spend") {
-				results = await
-					parseSpend(parsed, appList);
-				createOfferOptions.verbose && console.log(`created spend:${ contentType } offers`);
-			} else if (title === "Earn") {
-				results = await
-					parseEarn(parsed, contentType, appList, createOfferOptions);
-				createOfferOptions.verbose && console.log(`created earn:${ contentType } offers`);
-			} else {
-				throw new Error("Failed to parse " + parsed[0][0]);
+		const offerList: OfferData[] = require(path(join(offersDir, OFFERS_JSON_FILENAME)));
+		const offerContents: OfferContentData[] = require(path(join(offersDir, OFFERS_CONTENTS_JSON_FILENAME)));
+		const offerContentsDict = offerContents.reduce((acc: any, value) => {
+			acc[value.offer_id] = value;
+			return acc;
+		}, {});
+		const offerData: Array<OfferData & OfferContentData> = offerList.map(value => Object.assign(value, offerContentsDict[value.id]));
+		await Promise.all(offerData.map(async offer => {
```

Comment:
Don't you think a simple for of loop is more readable? We don't need efficiency here

Reply:
even better, a for...loop just for you